benchmarks/synchrobench/src/main/java/com/oath/oak/synchrobench/MyBuffer.java

```diff
@@ -50,49 +56,61 @@ public static MyBuffer deserialize(ByteBuffer inputBuffer, int inputPos) {
         inputPos += Integer.BYTES;
 
         MyBuffer ret = new MyBuffer(capacity);
-        OakIntBufferSerializer.copyBuffer(inputBuffer, inputPos, capacity / Integer.BYTES, ret.buffer, 0);
+        OakIntBufferSerializer.copyBuffer(inputBuffer, inputPos, capacity / Integer.BYTES, ret.buffer, DATA_POS);
         return ret;
     }
 
-    public static int calculateSerializedSize(MyBuffer object) {
-        return object.buffer.capacity() + Integer.BYTES;
-    }
-
     public static int compareBuffers(ByteBuffer buff1, int pos1, int cap1, ByteBuffer buff2, int pos2, int cap2) {
         return OakIntBufferComparator.compare(buff1, pos1, cap1 / Integer.BYTES,
             buff2, pos2, cap2 / Integer.BYTES);
     }
 
     public static int compareBuffers(ByteBuffer buffer1, ByteBuffer buffer2) {
-        return compareBuffers(buffer1, buffer1.position(), buffer1.capacity(),
-            buffer2, buffer2.position(), buffer2.capacity());
-    }
+        int pos1 = buffer1.position();
```

Comment:
Is it the main change revealing the bug?

Reply:
Yes :)